tests/ast/data/DataFrame.collect.test

```diff
@@ -36,11 +36,77 @@ df.collect_nowait(statement_params={"SF_PARTNER": "FAKE_PARTNER"}, log_on_except
 
 ## EXPECTED ENCODED AST
 
-Cj8KPQovggwsEgoKCAoGdGFibGUxGhoaFlNSQ19QT1NJVElPTl9URVNUX01PREUoGSICCAESBAoCZGYYASICCAEKMwoxCifSBiQIARABGgIIATIaGhZTUkNfUE9TSVRJT05fVEVTVF9NT0RFKBsSABgCIgIIAgoIEgYIAxICCAIQARoREg8KDQoFZmluYWwQAxgJIBIiBBABGBc=
-CjEKLwol0gYiEAEaAggBMhoaFlNSQ19QT1NJVElPTl9URVNUX01PREUoHRIAGAQiAggECggSBggFEgIIBBABGhESDwoNCgVmaW5hbBADGAkgEiIEEAEYFw==
-Ck8KTQpD0gZACAEQARoCCAEyGhoWU1JDX1BPU0lUSU9OX1RFU1RfTU9ERSgfOhoKClNGX1BBUlRORVISDEZBS0VfUEFSVE5FUhIAGAYiAggGCggSBggHEgIIBhABGhESDwoNCgVmaW5hbBADGAkgEiIEEAEYFw==
-Ck0KSwpB0gY+GgIIASABMhoaFlNSQ19QT1NJVElPTl9URVNUX01PREUoIToaCgpTRl9QQVJUTkVSEgxGQUtFX1BBUlRORVISABgIIgIICAoIEgYICRICCAgQARoREg8KDQoFZmluYWwQAxgJIBIiBBABGBc=
-CjMKMQon0gYkEAEaAggBKAEyGhoWU1JDX1BPU0lUSU9OX1RFU1RfTU9ERSgjEgAYCiICCAoKCBIGCAsSAggKEAEaERIPCg0KBWZpbmFsEAMYCSASIgQQARgX
-Ck8KTQpD0gZAEAEaAggBKAEyGhoWU1JDX1BPU0lUSU9OX1RFU1RfTU9ERSglOhoKClNGX1BBUlRORVISDEZBS0VfUEFSVE5FUhIAGAwiAggMCggSBggNEgIIDBABGhESDwoNCgVmaW5hbBADGAkgEiIEEAEYFw==
-Ck8KTQpD0gZAGgIIASABKAEyGhoWU1JDX1BPU0lUSU9OX1RFU1RfTU9ERSgnOhoKClNGX1BBUlRORVISDEZBS0VfUEFSVE5FUhIAGA4iAggOCggSBggPEgIIDhABGhESDwoNCgVmaW5hbBADGAkgEiIEEAEYFw==
-EAEaERIPCg0KBWZpbmFsEAMYCSASIgQQARgX
+body {
```

Comment:
We may want to consider stripping some fields from this representation. I.e., in the message comparison the python versions are ignored and a few other fields. I'd recommend removing them or setting them to a constant value for test purposes. Else, we'll end up with a lot of diffs for these fields that are unstable across machines.

The logic for this is in `test_ast` in the following lines:

```
 # Actual and expected may have been encoded by different client language versions, e.g. Python 3.8.10 and
            # Python 3.9.3. Make comparison here client-language agnostic by removing the data from the message.
            actual_message.ClearField("client_language")
            expected_message.ClearField("client_language")

            # Similarly, for create_dataframe with temporary tables clear them as they may differ from session to session.
            ClearTempTables(actual_message)
            ClearTempTables(expected_message)
```

Reply:
Updated